File: xpenbox-back/src/main/java/org/xpenbox/category/dto/CategoryResponseDTO.java
package org.xpenbox.category.dto;

import io.quarkus.runtime.annotations.RegisterForReflection;

/**
 * Data Transfer Object for responding with Category details.
 * @param resourceCode              the unique resource code of the category
 * @param name                      the name of the category
 * @param color                     the color associated with the category
 * @param lastUsedDateTimestamp     the timestamp of the last used date
 * @param usageCount                the usage count of the category
 * @param state                     the state of the category
 */
@RegisterForReflection
public record CategoryResponseDTO (
    String resourceCode,
    String name,
    String color,
    Long lastUsedDateTimestamp,
    Long usageCount,
    Boolean state
) { }

File: xpenbox-front/src/app/feature/category/model/category.response.dto.ts
/**
 * DTO representing the response structure for a category.
 * Includes resource code, name, color, amount, budget, hasBudget, last used date timestamp, usage count, and state.
 */
export interface CategoryResponseDTO {
    resourceCode: string;
    name: string;
    color: string;
    amount: number;
    budget: number;
    hasBudget: boolean;
    lastUsedDateTimestamp: number;
    usageCount: number;
    state: boolean;
}
